pkg/scheduler/scheduler.go

```diff
@@ -405,7 +405,7 @@ func (s *Scheduler) cancelRequestAndRemoveFromPending(key queue.RequestKey, reas
 
 	req := s.schedulerInflightRequests[key]
 	if req != nil {
-		req.CancelFunc(cancellation.NewErrorf(reason))
+		req.CancelFunc(cancellation.NewErrorf("%s", reason))
```

Comment:
```suggestion
		req.CancelFunc(cancellation.NewError(reason))
```

Reply:
I had to reverse this one, as `reason` is a string, and `cancellation.NewError()` takes an inner error instead. It seems %s is still more appropriate here

Reply:
I forgot it takes an error.

Reply:
[Nit]
```suggestion
		req.CancelFunc(cancellation.NewError(errors.New(reason)))
```